File: src/main/java/com/urlshortner/urlshortner/Controller/CounterController.java
package com.urlshortner.urlshortner.Controller;

import com.urlshortner.urlshortner.Model.OperationResult;
import com.urlshortner.urlshortner.Service.CounterService;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RestController;

@RestController
@RequestMapping(path = "test")
public class CounterController {

    @Autowired
    CounterService counterService;

    @GetMapping("")
    String test() {
        OperationResult<Void, String> result =
                counterService.insertCounterRange(1, 5);
        if (result.isFailed()) {
            return result.getFailureData();
        }
        return "Success";
    }

    @GetMapping("get")
    String getCounterValue() {
        OperationResult<Long, String> result = counterService.getCounterValue();
        if(result.isFailed()) {
            return result.getFailureData();
        } else if (result.isRangeExhausted()) {
            return "Reset: " + resetCounterValue();
        }
        return "Counter Value: " + result.getSuccessData();
    }

    @GetMapping("reset")
    String resetCounterValue() {
        OperationResult<Void, String> result = counterService.resetCounter(6, 10);
        if (result.isFailed()) {
            return result.getFailureData();
        }
        return "Success";
    }

}

File: src/main/java/com/urlshortner/urlshortner/controller/CounterController.kt
package com.urlshortner.urlshortner.controller

import com.urlshortner.urlshortner.model.OperationResult
import com.urlshortner.urlshortner.service.ShortIdService
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.web.bind.annotation.GetMapping
import org.springframework.web.bind.annotation.RequestMapping
import org.springframework.web.bind.annotation.RestController

@RestController
@RequestMapping(path = ["test"])
class CounterController {

    @Autowired
    lateinit var shortIdService: ShortIdService

    @get:GetMapping("get")
    val counterValue: String?
        get() {
            return when (val result = shortIdService.getShortId()) {
                is OperationResult.Failure -> result.reason
                is OperationResult.Success -> result.data
            }
        }
}
